### PR TITLE
aealite fixes: p2p acn support, tls support

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

[This trello ticket](https://trello.com/c/uqEwaxwy/797-port-https-githubcom-fetchai-agents-aea-pull-2665)
 - added a test that ensures aealite type definition and .proto file in aealite is identical to `packages/valory/protocols/acn`

comes after #222  and #225 

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: libs/go/aealite/connections/tcpsocket.go [hunk 1]
```diff
@@ -21,25 +21,63 @@
...
+	port          uint16
+	conn          *tls.Conn
+	peerPublicKey string
 }
 
 func (sock *TCPSocketChannel) Connect() error {
 	var err error
-	sock.conn, err = net.Dial("tcp", sock.address+":"+strconv.FormatInt(int64(sock.port), 10))
+	conf := &tls.Config{
+		InsecureSkipVerify: true,
```
[the remark this answers is not in the document]
Comment (Karrenbelt) [reply libs/go/aealite/connections/tcpsocket.go:44]: Interesting to take note of. Not sure why, or even if this is needed.

**Comment by DavidMinarsch** [comment]
Why not remove aealite entirely from this repo? It's a pain to develop here and atm we don't develop it anyway.

**Comment by Karrenbelt** [comment]
> Why not remove aealite entirely from this repo? It's a pain to develop here and atm we don't develop it anyway.

That is fine with me. It was an ACN related ticket on the Trello board, which is why I decided to attempt to quickly cover it before heading to Montreux.

File: libs/go/aealite/connections/p2pclient.go [hunk 2]
```diff
@@ -122,13 +127,20 @@ func (client *P2PClientApi) InitFromEnv(envFile string) error {
...
 	client.clientConfig = &P2PClientConfig{host: host, port: uint16(portConv)}
 
-	client.socket = NewSocket(client.clientConfig.host, client.clientConfig.port)
+	client.socket = NewSocket(client.clientConfig.host, client.clientConfig.port, client.agentRecord.PeerPublicKey)
 	client.initialised = true
 	return nil
 }
 
 func (client *P2PClientApi) Put(envelope *protocols.Envelope) error {
-	return writeEnvelope(client.socket, envelope)
```
Comment (DavidMinarsch) [inline libs/go/aealite/connections/p2pclient.go:131]: why not keep it nicely wrapped away?

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**